### PR TITLE
Runtime warning docs

### PR DESCRIPTION
**Pull request by justinfagnani**
I realized we didn't mention these in the docs at all yet.

These aren't in great shape, but I wanted to put up the raw information in a PR at least.

### CODE REVIEW

**Comment by lazka** [comment]
Does it state somewhere if these warnings are enabled by default? It's not clear to me based on this text.

**Comment by AndrewJakubowicz** [comment]
@lazka From this [section on Development and Production builds](https://lit.dev/docs/tools/development/#development-and-production-builds), I believe you must opt into the development build.

File: packages/lit-dev-content/site/docs/tools/development.md [hunk 1]
```diff
@@ -37,6 +37,44 @@ export default {
...
+
+The development builds of ReactiveElement and LitElement support extra runtime warnings that can help identify issues that would be costly to check for in production builds.
+
+There is an API to enable and disable some warnings, based on the kind of warning. There are two warning kinds that can be enabled and disable currently:
+* `'migration'`: warnings related to migration from LitElement 2.x
+* `'change-in-update'`: warnings related to changing reactive state during an update
+
+The methods to control warnings are `ReactiveElement.disableWarning()` and `ReactiveElement.enableWarning()`. You can call them on any subclass of ReactiveElement, including LitElement and your own classes, and the warnings will be enabled or disabled for only those classes. For instance, you can disable warnings on all ReactiveElement classes, all LitElement classes, or only specific LitElement subclasses.
+
+These methods are only available in development builds, so be sure to guard their access. We reccomend using optional chaining.
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/site/docs/tools/development.md:50]: nitpick - typo
```suggestion
These methods are only available in development builds, so be sure to guard their access. We recommend using optional chaining.
```

File: packages/lit-dev-content/site/docs/tools/development.md [hunk 2]
```diff
@@ -37,6 +37,44 @@ export default {
...
+
+Examples:
+```ts
+import {LitElement, ReactiveElement} from 'lit';
+
+// Disable migration warnings on all ReactiveElements,
+// including LitElements
+ReactiveElement.disableWarning.?('migration');
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/site/docs/tools/development.md:58]: typo - optional chaining call
```suggestion
ReactiveElement.disableWarning?.('migration');
```

File: packages/lit-dev-content/site/docs/tools/development.md [hunk 3]
```diff
@@ -37,6 +37,44 @@ export default {
...
+
+// Disable update warnings on all LitElements
+LitElement.disableWarning.?('change-in-update');
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/site/docs/tools/development.md:61]:
```suggestion
LitElement.disableWarning?.('change-in-update');
```

File: packages/lit-dev-content/site/docs/tools/development.md [hunk 4]
```diff
@@ -37,6 +37,44 @@ export default {
...
+
+// Disable update warnings on one element
+MyElement.disableWarning.?('change-in-update');
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/site/docs/tools/development.md:64]:
```suggestion
MyElement.disableWarning?.('change-in-update');
```

**Review by AndrewJakubowicz — COMMENTED** [review]
Excellent! Left some minor syntax comments.

**Comment by lazka** [comment]
> @lazka From the section above on [Development and Production builds](https://lit.dev/docs/tools/development/#development-and-production-builds), I believe you must opt into the development build.

It also doesn't state if the warnings are enabled by default or not in the development build.

File: packages/lit-dev-content/site/docs/tools/development.md [hunk 5]
```diff
@@ -37,6 +37,44 @@ export default {
 };
 ```
 
+### Development build runtime warnings
+
+The development builds of ReactiveElement and LitElement support extra runtime warnings that can help identify issues that would be costly to check for in production builds.
+
+There is an API to enable and disable some warnings, based on the kind of warning. There are two warning kinds that can be enabled and disable currently:
+* `'migration'`: warnings related to migration from LitElement 2.x
+* `'change-in-update'`: warnings related to changing reactive state during an update
```
Comment (arthurevans) [inline packages/lit-dev-content/site/docs/tools/development.md:46]:
```suggestion
* `'migration'`. Warnings related to migration from LitElement 2.x. Off by default.
* `'change-in-update'`. Warnings related to changing reactive state during an update. On by default.
```

File: packages/lit-dev-content/site/docs/tools/development.md [hunk 6]
```diff
@@ -37,6 +37,44 @@ export default {
...
+
+The methods to control warnings are `ReactiveElement.disableWarning()` and `ReactiveElement.enableWarning()`. You can call them on any subclass of ReactiveElement, including LitElement and your own classes, and the warnings will be enabled or disabled for only those classes. For instance, you can disable warnings on all ReactiveElement classes, all LitElement classes, or only specific LitElement subclasses.
+
+These methods are only available in development builds, so be sure to guard their access. We recommend using optional chaining.
+
+Examples:
+```ts
+import {LitElement, ReactiveElement} from 'lit';
+
+// Disable migration warnings on all ReactiveElements,
```
Comment (arthurevans) [inline packages/lit-dev-content/site/docs/tools/development.md:56]:
```suggestion
// Turn off migration warnings on all ReactiveElements,
```

File: packages/lit-dev-content/site/docs/tools/development.md [hunk 7]
```diff
@@ -37,6 +37,44 @@ export default {
...
+// including LitElements
+ReactiveElement.disableWarning?.('migration');
+
+// Disable update warnings on all LitElements
```
Comment (arthurevans) [inline packages/lit-dev-content/site/docs/tools/development.md:60]:
```suggestion
// Turn off update warnings on all LitElements
```

File: packages/lit-dev-content/site/docs/tools/development.md [hunk 8]
```diff
@@ -37,6 +37,44 @@ export default {
...
+LitElement.disableWarning?.('change-in-update');
+
+// Disable update warnings on one element
```
Comment (arthurevans) [inline packages/lit-dev-content/site/docs/tools/development.md:63]:
```suggestion
// Turn off update warnings on one element
```

File: packages/lit-dev-content/site/docs/tools/development.md [hunk 9]
```diff
@@ -37,6 +37,44 @@ export default {
...
+MyElement.disableWarning?.('change-in-update');
+
+```
+
+You can also control warnings within a single class with the `static enabledWarnings`:
```
Comment (arthurevans) [inline packages/lit-dev-content/site/docs/tools/development.md:68]: Not 100% sure this is sounds good but the previous one isn't grammatical. (Also not sure whether having another way to do this really adds much utility.)

```suggestion
You can also control warnings within a single class by defining a `static enabledWarnings` property:
```

**Review by arthurevans — CHANGES_REQUESTED** [review]
Suggesting some edits.

**Comment by justinfagnani** [comment]
@arthurevans PTAL

File: packages/lit-dev-content/site/docs/tools/development.md [hunk 10]
```diff
@@ -37,6 +37,45 @@ export default {
 };
 ```
 
+### Development build runtime warnings
+
+The development builds of `ReactiveElement` and `LitElement` support extra runtime warnings that can help identify issues that would be costly to check for in production builds.
+
+Some warnings are always displayed.  There are also two categories of _optional warnings_ that can be turned on or off:
+:
```
Comment (arthurevans) [inline packages/lit-dev-content/site/docs/tools/development.md:45]:
```suggestion

```

Double colon here.

**Review by arthurevans — APPROVED** [review]
Made some last fixes, LGTM.

**Pull request merged**